packages/sitecore-jss/src/layout/utils.ts

```diff
@@ -113,7 +113,13 @@ export function isFieldValueEmpty(field: GenericFieldValue | Partial<Field>): bo
     !(fieldValue as { [key: string]: unknown }).src;
   const isLinkFieldEmpty = (fieldValue: GenericFieldValue) =>
     !(fieldValue as { [key: string]: unknown }).href;
-  const isDateFieldEmpty = (fieldValue: GenericFieldValue) => fieldValue === EMPTY_DATE_FIELD_VALUE;
+  const isDateFieldEmpty = (fieldValue: GenericFieldValue) => {
+    if (typeof fieldValue === 'string') {
+      return fieldValue === EMPTY_DATE_FIELD_VALUE;
+    } else {
+      return !(typeof (fieldValue as Date)?.getMonth === 'function' && !isNaN((fieldValue as Date)?.getMonth()));
```

Comment:
Do we need to verify this case?
Is it even possible to create a _Date_ instance and it will produce an invalid value? When you create _Date_ you have an initial value in any case.

Reply:
It is an edge case, but I think yes: if user is passing Date field directly (and its not coming from sitecore, in which case it will be string), potentially it can be instantiated with invalid date string.

Reply:
Makes sense to me, it's clear now